extensions/amp-subscriptions/0.1/local-subscription-platform.js

```diff
@@ -110,6 +130,40 @@ export class LocalSubscriptionPlatform {
     return this.readerIdPromise_;
   }
 
+  /**
+   * Add event listener for the subscriptions action
+   * @private
+   */
+  initializeListeners_() {
+    this.document_.addEventListener('click', e => {
+      const element = closestBySelector(e.target, '[subscriptions-action]');
+      if (element) {
+        const action = element.getAttribute('subscriptions-action');
+        this.executeAction(action);
+      }
+    });
+  }
+
+  /**
+   * Renders the platform specific UI
+   */
+  render() {
+    this.renderer_.render();
+  }
+
+  /**
+   * Executes action for the local platform.
+   * @param {string} action
+   */
+  executeAction(action) {
+    const actionExecution = this.actions_.execute(action);
+    actionExecution.then(result => {
+      if (result) {
+        this.getEntitlements();
```

Comment:
Would this have any affect on rendering or such? Also, you may want to execute this call via the top-level service b/c changing of entitlements here could result in selecting a new platform... Maybe something like: `service.reAuthorizePlatform(this)` or such?

Reply:
yes, added a todo which will resolve with the actio delegation PR as both need a service interface